src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -4240,13 +4240,13 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
     </Copy>
 
-    <!-- If this project produces reference assemblies *and* copied (possibly transitive)
+    <!-- If this project produces reference assemblies *or* copied (possibly transitive)
          references on this build, subsequent builds of projects that depend on it must
          not be considered up to date, so touch this marker file that is considered an
          input to projects that reference this one. -->
     <Touch Files="$(CopyUpToDateMarker)"
            AlwaysCreate="true"
-           Condition="'$(ProduceReferenceAssembly)' == 'true' and '@(ReferencesCopiedInThisBuild)' != ''" />
+           Condition="'$(ProduceReferenceAssembly)' == 'true' or '@(ReferencesCopiedInThisBuild)' != ''" />
```

Comment:
Oh, ok. The design process of this feature has kinda been a big game of Telephone, so I came away with a different understanding of how this situation would be dealt with but reading the design again, I see what I missed. OK, I'll roll back to the original fix.